src/widgets/OverlayWindow.cpp

```diff
@@ -0,0 +1,529 @@
+    if (!seq.isEmpty())
+    {
+        return {seq, false};
+    }
+    seq = getApp()->getHotkeys()->getDisplaySequence(
+        HotkeyCategory::Split, u"togglePopupInertia"_s, {{u"thisOrAll"_s}});
+    if (!seq.isEmpty())
+    {
+        return {seq, false};
+    }
+    return {
+        getApp()->getHotkeys()->getDisplaySequence(HotkeyCategory::Split,
+                                                   u"togglePopupInertia"_s),
+        true,
+    };
+}
+
+}  // namespace
+
+namespace chatterino {
+
+using namespace std::chrono_literals;
+
+OverlayWindow::OverlayWindow(IndirectChannel channel)
+    : QWidget(nullptr,
+              Qt::Window | Qt::FramelessWindowHint | Qt::WindowStaysOnTopHint)
+#ifdef Q_OS_WIN
+    , sizeAllCursor_(::LoadCursor(nullptr, IDC_SIZEALL))
+#endif
+    , channel_(std::move(channel))
+    , channelView_(nullptr)
+    , interaction_(this)
+{
+    this->setAttribute(Qt::WA_DeleteOnClose);
+    this->setWindowTitle(u"Chatterino - Overlay"_s);
+
+    // QGridLayout is (ab)used to stack widgets and position them
+    auto *grid = new QGridLayout(this);
+    grid->addWidget(&this->channelView_, 0, 0);
+    this->interaction_.attach(grid);
+#ifndef OVERLAY_NATIVE_MOVE
+    grid->addWidget(new InvisibleSizeGrip(this), 0, 0,
+                    Qt::AlignBottom | Qt::AlignRight);
+#endif
+
+    // the interaction overlay currently captures all events
+    this->interaction_.installEventFilter(this);
+
+    this->shortInteraction_.setInterval(750ms);
+    QObject::connect(&this->shortInteraction_, &QTimer::timeout, [this] {
+        this->endInteraction();
+    });
+
+    this->channelView_.installEventFilter(this);
+    this->channelView_.setChannel(this->channel_.get());
+    this->channelView_.setIsOverlay(true);  // use overlay colors
+    this->channelView_.setAttribute(Qt::WA_TranslucentBackground);
+    this->holder_.managedConnect(this->channel_.getChannelChanged(), [this]() {
+        this->channelView_.setChannel(this->channel_.get());
+    });
+    this->channelView_.scrollbar()->setShowThumb(false);
+
+    this->setAutoFillBackground(false);
+    this->resize(300, 500);
+    this->move(QCursor::pos() - this->rect().center());
+    this->setContentsMargins(0, 0, 0, 0);
+    this->setAttribute(Qt::WA_TranslucentBackground);
+
+    auto *settings = getSettings();
+    settings->enableOverlayShadow.connect(
+        [this](bool value) {
+            if (value)
+            {
+                this->dropShadow_ = new QGraphicsDropShadowEffect;
+                this->channelView_.setGraphicsEffect(this->dropShadow_);
+            }
+            else
+            {
+                this->channelView_.setGraphicsEffect(nullptr);
+                this->dropShadow_ = nullptr;  // deleted by setGraphicsEffect
+            }
+            this->applyTheme();
+        },
+        this->holder_);
+    settings->overlayBackgroundOpacity.connect(
+        [this] {
+            this->channelView_.updateColorTheme();
```

Comment:
Missing a call to `update` (or something similar) to ensure the opacity updates as it's being changed
```suggestion
            this->channelView_.updateColorTheme();
            this->update();
```